map.go

```diff
@@ -17,6 +17,7 @@ import (
 var (
 	ErrKeyNotExist      = errors.New("key does not exist")
 	ErrKeyExist         = errors.New("key already exists")
+	ErrMapSize          = errors.New("map too small for given key")
```

Comment:
Express it the other way round: `key too big for map`? Why does this need to be exported?

Reply:
True, makes sense. I provided the rationale in the commit message: out-of-bounds writes return a message like `update failed: argument list too long`, which I find a bit misleading. It may be easier for us to figure out what this means by being familiar with the API, but I can imagine this being rather confusing when it happens to a user for the first time.

I'll go with your suggestion below of just adding some string context. We don't really need to `ErrMapSize` here, just thought I'd stick with the existing convention.